validation/immutable-references.go

```diff
@@ -0,0 +1,58 @@
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// ContractASTsWithImmutableReferences stores the `immutableReferences`, copied from the compiled contract AST (from the combined JSON
+// artifact from the monorepo. We do this because the contracts and compiled artifacts are not available in the superchain
+// registry. Ex: ethereum-optimism/optimism/packages/contracts-bedrock/forge-artifacts/MIPS.sol/MIPS.json
+var ContractASTsWithImmutableReferences = map[string]string{
+	"AnchorStateRegistryProxy": `"immutableReferences":{"92829":[{"start":387,"length":32},{"start":828,"length":32},{"start":2296,"length":32}]}`,
+	"DelayedWETHProxy":         `"immutableReferences":{"97827":[{"start":831,"length":32},{"start":4133,"length":32}]}`,
+	"FaultDisputeGame":         `"immutableReferences":{"92829":[{"start":387,"length":32},{"start":828,"length":32},{"start":2296,"length":32}]}`,
+	"MIPS":                     `"immutableReferences":{"85798":[{"start":178,"length":32},{"start":1771,"length":32}]}`,
+}
+
+// ImmutableReference to store the start/length of a contract's immutable references. Immutables
+// are directly assigned values once or are initialized in a constructor. They are encoded in the contract bytecode,
+// and the AST that is generated by the compiler has info about the offset & length within the bytecode.
+type ImmutableReference struct {
+	Start  int `json:"start"`
+	Length int `json:"length"`
+}
+
+// Define a struct to represent the structure of the JSON data
+type DeployedBytecode struct {
+	Bytecode            []byte                          `json:"object"`
+	ImmutableReferences map[string][]ImmutableReference `json:"immutableReferences"`
+}
```

Comment:
Nit: I know this is how the data is arranged in the solidity compiler output. However, I think a name like `bytecodeAndImmutableReferences` or something like that might be more informative?